src/main/java/com/greenfox/tribesoflagopus/backend/controller/BuildingController.java

```diff
@@ -0,0 +1,115 @@
+package com.greenfox.tribesoflagopus.backend.controller;
+
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingLevelInputDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingListDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.BuildingTypeInputDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.JsonDto;
+import com.greenfox.tribesoflagopus.backend.model.dto.StatusResponse;
+import com.greenfox.tribesoflagopus.backend.service.BuildingService;
+import com.greenfox.tribesoflagopus.backend.service.ErrorService;
+import com.greenfox.tribesoflagopus.backend.service.TokenService;
+import com.greenfox.tribesoflagopus.backend.service.UserService;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+
+@CrossOrigin("*")
+@RestController
+public class BuildingController {
+
+  private final BuildingService buildingService;
+  private final UserService userService;
+  private final ErrorService errorService;
+  private final TokenService tokenService;
+
+  @Autowired
+  public BuildingController(
+      BuildingService buildingService,
+      UserService userService,
+      ErrorService errorService, TokenService tokenService) {
+    this.buildingService = buildingService;
+    this.userService = userService;
+    this.errorService = errorService;
+    this.tokenService = tokenService;
+  }
+
+  @GetMapping("/kingdom/buildings")
+  public ResponseEntity<JsonDto> getListOfBuildings(
+      @RequestHeader(value = "X-tribes-token") String token) {
+
+    Long userId = tokenService.getIdFromToken(token);
+    if (userId == null) {
```

Comment:
There shouldn't be such case when you have a valid token, but the userId is null inside.
Instead of this you should check if the token is a valid token or not (is it in your db or not).